lib/search.js

```diff
@@ -43,7 +43,7 @@ function search(root, expression, maxDepth) {
     var value = toString(child)
     var id = child.data && child.data.hProperties && child.data.hProperties.id
 
-    if (parent !== root) {
+    if (parent !== root && parentTypeWhiteList.indexOf(parent.type) < 0) {
```

Comment:
Could you use [`unist-util-is`](https://github.com/syntax-tree/unist-util-is#istest-node-index-parent-context) here? That way the test is much more flexible, and it’s already included in `unist-util-visit` so it doesn’t add a new dependency.

Also: `parents` could be used for the variable name, right?